### PR TITLE
Migrate module configurations in RecoTracker{IterativeTracking} to use default cfipython

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Optimization of the python configurations: Improve maintainability by cleaning up the duplicated and cloning from the default/reference configurations.
 
In this PR, 6 files changed.  
RecoTracker/IterativeTracking/python/ElectronSeeds_cff.py
RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py
RecoTracker/IterativeTracking/python/MixedTripletStep_cff.py
RecoTracker/IterativeTracking/python/PixelLessStep_cff.py
RecoTracker/IterativeTracking/python/PixelPairStep_cff.py
RecoTracker/IterativeTracking/python/TobTecStep_cff.py

(The previous PRs were  [PR#33207](https://github.com/cms-sw/cmssw/pull/33207), [PR#33307](https://github.com/cms-sw/cmssw/pull/33307), [PR#33352](https://github.com/cms-sw/cmssw/pull/33352), [PR#33543](https://github.com/cms-sw/cmssw/pull/33543),  [PR#33563](https://github.com/cms-sw/cmssw/pull/33563),  [PR#33671](https://github.com/cms-sw/cmssw/pull/33671), [PR#33901](https://github.com/cms-sw/cmssw/pull/33901))

- commit 1 : Replace explicit configuration with a reference from cfipython/. (migrating EDProducer("type", .. -> typeDefault.clone()) 
- commit 2 : Remove the type specifications already presented in cfipython/fillDescriptions reference for improved syntax safety.

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_12_0_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py [hunk 1]
```diff
@@ -76,9 +75,6 @@
     vertexSrc      = 'firstStepGoodPrimaryVertices',
     howToUseMeasurementTracker = 'Never'
 ))
-jetCoreRegionalStepEndcapTrackingRegions = jetCoreRegionalStepTrackingRegions.clone(RegionPSet=dict(
```
Comment (perrotta) [inline RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py:79]: I miss most of the cleanings in this file: I'm not saying they are not correct, but could you please explain the logic behind? Are all those removed modules/PSets really not needed (or replaced somehow)?

File: RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py [hunk 2]
```diff
@@ -7,10 +7,7 @@
 # This step runs over all clusters
 
 # run only if there are high pT jets
-jetsForCoreTracking = cms.EDFilter('CandPtrSelector', src = cms.InputTag('ak4CaloJetsForTrk'), cut = cms.string('pt > 100 && abs(eta) < 2.5'), filter = cms.bool(False))
-
-jetsForCoreTrackingBarrel = jetsForCoreTracking.clone( cut = 'pt > 100 && abs(eta) < 2.5' )
-jetsForCoreTrackingEndcap = jetsForCoreTracking.clone( cut = 'pt > 100 && abs(eta) > 1.4 && abs(eta) < 2.5' )
+jetsForCoreTracking = cms.EDFilter('CandPtrSelector', src = cms.InputTag('ak4CaloJetsForTrk'), cut = cms.string('pt > 100 && abs(eta) < 2.5'))
```
[the remark this answers is not in the document]
Comment (slava77) [reply RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py:10]: it looks like the changes revert #33531 .
Perhaps this is a result of a not very successful rebase?

**Comment by perrotta** [comment]
@jeongeun please redo  RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py based on a recent IB. As Slava already [pointed out](https://github.com/cms-sw/cmssw/pull/33903#discussion_r645148010) you probably based your code changes on a IB older than CMSSW_12_0_X_2021-05-24-1100, which is when #33531 was merged.

File: RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py (hunk 1 above)
Comment (jeongeun) [reply RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py:79]: > @jeongeun please redo RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py based on a recent IB. As Slava already [pointed out](https://github.com/cms-sw/cmssw/pull/33903#discussion_r645148010) you probably based your code changes on a IB older than CMSSW_12_0_X_2021-05-24-1100, which is when #33531 was merged.

@perrotta Thanks for your comment. I did rebase to move this branch to the recent IB (120X_06-07-2300), but it's strange that nothing different (no conflict) was found. The #33531 update was not applied properly I think..

Comment (jeongeun) [reply RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py:79]: Now it updated to pr #34007.

**Pull request closed without merging**